R/utils-ui.R

```diff
@@ -196,36 +196,31 @@ message <- function(...) {
     instead of {.fun message}")
 }
 
-warn_for_verbose <- function(verbose = TRUE, env = caller_env()) {
+warn_for_verbose <- function(verbose = TRUE,
+                             env = caller_env(),
+                             user_env = caller_env(2)) {
   # this is about whether `verbose` was present in the **user's** call to the
   # calling function
-  # don't worry about the `verbose = TRUE` default here
+  # don't worry about the `verbose = TRUE` default in warn_for_verbose()
   if (!lifecycle::is_present(verbose) || isTRUE(verbose)) {
     return(invisible())
   }
 
-  called_from <- sys.parent(1)
-  if (called_from == 0) {
-    # called from global env, presumably in a test or during development
-    caller <- "some_googledrive_function"
-  } else {
-    called_as <- sys.call(called_from)
-    if (is.call(called_as) && is.symbol(called_as[[1]])) {
-      caller <- as.character(called_as[[1]])
-    } else {
-      caller <- "some_googledrive_function"
-    }
-  }
   lifecycle::deprecate_warn(
     when = "2.0.0",
-    what = glue("{caller}(verbose)"),
+    what = I(cli::format_inline("The {.arg verbose} argument")),
```

Comment:
I think it would be fine to just use ``"The `verbose` argument"``; using cli in this way is rather unusual so I wouldn't want to feature it too prominently. 

I'd almost prefer keeping `caller` and then doing `what = paste0(caller, "(verbose)")`.

Reply:
But my impression is that I won't know `caller` without my DIY gymnastics, which I'm removing. I'm not aware of any clean way to get that with simple usage of lifecycle. Do you think I'm missing something?

Reply:
(I went ahead and removed the use of `cli::format_inline()`.)

Reply:
Yeah, I meant keep `caller` (or make it an explicit argument to pass along)

Reply:
I don't think we want to subject R Packages readers to my code for learning `caller`, so the version I put in the book will be a simple statement about `verbose`. But maybe, before I actually merge this, I'll put the `caller` stuff back.

This is PR is also to make sure that whatever I put in the book actually works, even if it's not what I ultimately use.

Reply:
I thought about the explicit `caller` at the very start, but it would be pretty annoying in this particular case.